.github/workflows/release.yml

```diff
@@ -47,60 +54,69 @@ jobs:
           minor="$(./mvnw build-helper:parse-version help:evaluate -Dexpression=parsedVersion.minorVersion -q -DforceStdout)"
           patch="$(./mvnw build-helper:parse-version help:evaluate -Dexpression=parsedVersion.incrementalVersion -q -DforceStdout)"
           qualifier="$(./mvnw build-helper:parse-version help:evaluate -Dexpression=parsedVersion.qualifier -q -DforceStdout)"
-          
+
           case $bump in
           hotfix)
-            branch_name="$(git branch --show-current)"
+            branch_name="release/${major}.${minor}"
             release_version="${major}.${minor}.${patch}"
             release_development_version="${major}.${minor}.$((patch + 1))-${qualifier}"
+            # master_development_version cannot be computed here, nor it is needed as this will run on release branch
             ;;
           major)
             new_major=$((major + 1))
             branch_name="release/${new_major}.0"
             release_version="${new_major}.0.0"
             release_development_version="${new_major}.0.1-${qualifier}"
+            master_development_version="${new_major}.1.0-${qualifier}"
             ;;
           minor)
             new_minor=$((minor + 1))
-            branch_name="release/${major}.${new_minor}"
-            release_version="${major}.${new_minor}.0"
+            branch_name="release/${major}.${minor}"
+            release_version="${major}.${minor}.0"
```

Comment:
Release version is `${major}.${minor}.${patch}` because we should expect `master` branch already be on version `x.(y+1).0-SNAPSHOT` when latest released version is `x.y.z`